clevertap-core/src/main/java/com/clevertap/android/sdk/Utils.java

```diff
@@ -179,6 +189,94 @@ public static Bitmap getBitmapFromURL(@NonNull String srcUrl) {
         }
     }
 
+    public static Bitmap getBitmapFromURLWithSizeConstraint(String srcUrl, int size) {
+        // Safe bet, won't have more than three /s
+        srcUrl = srcUrl.replace("///", "/");
+        srcUrl = srcUrl.replace("//", "/");
+        srcUrl = srcUrl.replace("http:/", "http://");
+        srcUrl = srcUrl.replace("https:/", "https://");
+        HttpURLConnection connection = null;
+        try {
+            URL url = new URL(srcUrl);
+            connection = (HttpURLConnection) url.openConnection();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected a URL in your code that uses an unencrypted HTTP protocol connection. This might expose sensitive data to third parties. To prevent this, we recommend that you construct all URLs using the secure HTTPS protocol.

[Learn more](https://owasp.org/www-community/vulnerabilities/Insecure_Transport)